tests/MongoDB.Driver.Tests/AggregateFluentTests.cs

```diff
@@ -1031,6 +1031,80 @@ public void ReplaceWith_should_add_the_expected_stage([Values(false, true)] bool
             }
         }
 
+        public class Item1
+        {
+            public ObjectId Id { get; set; }
+            public string Name1 { get; set; }
+        }
+
+        public class Item2
+        {
+            public ObjectId Id { get; set; }
+            public string Name2 { get; set; }
+        }
+
+        [SkippableFact]
+        [ParameterAttributeData]
+        public void UnionWith_with_different_schemas_and_projection_should_return_the_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateUnionWith);
+
+            var databaseName = "test";
+            var items1CollectionName = "items1";
+            var items2CollectionName = "items2";
+
+            var client = CreateClient();
+            DropCollection(client, databaseName, items1CollectionName);
+            DropCollection(client, databaseName, items2CollectionName);
+
+            var db = client.GetDatabase(databaseName);
+            var items1Collection = db.GetCollection<Item1>(items1CollectionName);
+            var items2Collection = db.GetCollection<Item2>(items2CollectionName);
+
+            var item1Documents = new[]
+            {
+                new Item1 { Name1 = "almonds" },
+                new Item1 { Name1 = "cookies" }
+            };
+            items1Collection.InsertMany(item1Documents);
+
+            var item2Documents = new[]
+            {
+                new Item2 { Name2 = "cookies" },
+                new Item2 { Name2 = "cookies" },
+                new Item2 { Name2 = "pecans" }
+            };
+            items2Collection.InsertMany(item2Documents);
+
+            var withPipeline =
+                new EmptyPipelineDefinition<Item2>()
+                .Match(x => x.Name2 == "cookies")
+                .Project(
+                    item2 => 
+                        new Item1
+                        {
+                            Name1 = item2.Name2 // manually project Item2 into Item1
+                        });
```

Comment:
Let's convert `Item2` instances to `ItemResult` instances:

    var withPipeline =
        new EmptyPipelineDefinition<Item2>()
        .Match(x => x.Name2 == "cookies")
        .Project(x => new ItemResult { Name = x.Name2 });

Note: the call to `Project` fits nicely on one line, no need to spread it across 6 lines